docs/content/querying/limitspec.md

```diff
@@ -13,9 +13,19 @@ The default limit spec takes a limit and the list of columns to do an orderBy op
     "type"    : "default",
     "limit"   : <integer_value>,
     "columns" : [list of OrderByColumnSpec],
+    "limitGranularity" : [QueryGranularity],
+    "limitPerGroup: <boolean_value>
 }
 ```
 
+|property|description|required?|
+|--------|-----------|---------|
+|type|This String should always be "default". |yes|
+|limit|Number of limit.|yes|
+|columns|Ordering to apply limit; see [OrderByColumnSpec]. |no|
+|limitGranularity|Granularity for limit. see [Granularities](../querying/granularities.html).|no|
+|limitPerGroup|Whether to apply limit to each group; User can possibly get up to <limit> * <number of groups> rows. |no|
```

Comment:
I am looking at this PR from the perspective of a user (i.e. only looking at the doc files and not at the code).

I do not understand what `limitGranularity` is. Could you add an example? When does it make sense to have it different from the granularity in the query? Also in the JSON example it appears to be an array so should it be called `limitGranularities`? An example here is defiantly needed.

`limitPerGroup` is cool would it prevent the `groupBy` from crashing if it is asked to materialize more than 500000 results? Like if I do a huge groupBy but set `limitPerGroup` to a low number would it always work?

It is not clear to me how this issue relates to #1926 . The problem in #1926 was that the sort was not getting applied across the groups does this PR fix that?

Finally when I first read these new docs I assumed `limitPerGroup` was a number and that `limitPerGroup` was applied per group and then the global `limit` was applied also. I envision that as being more useful. As in I think it is useful to still have a `limit` even if there is a limit to every group.